sw/device/lib/dif/README.md

```diff
@@ -50,6 +51,30 @@ there are some relaxations of these rules for them described at the end.
 * DIFs must use enum return codes rather than booleans for reporting errors. If
   a DIF can either error or instead produce a value, it should return an enum
   return code, and use an out-parameter for returning the produced value.
+  * DIF libraries must have a general return code enum called
+    `dif_<ip>_result_t`. It must define the following constants (in this order):
+    * `kDif<ip>Ok`, with value 0, to denote the call succeeded.
+    * `kDif<ip>Error`, to denote a non-specific error happened during the call.
+      This is for the `default:` case of enum switches (as noted below), and for
+      assertion errors (usually where the function has already caused side
+      effects so `kDif<ip>BadArg` cannot be used).
+    * `kDif<ip>BadArg`, to denote that the caller supplied incorrect arguments.
+      This value must only be returned before the function does anything that
+      causes side effects.
+  * Other return code enums can be defined for functions that fail in more
+    specific ways, and these other return code enums can be shared between
+    multiple functions that fail in the same way.
+    These more specific return codes must be named
+    `dif_<ip>_<operation>_result_t`, their constants should be prefixed with
+    `kDif<ip><operation>`, and the first three constants should be the same as
+    the three constants in `dif_<ip>_result_t`.
+    The additional, specific return code constants must all be defined after
+    the three general constants.
+  * DIFs must document the meaning of all error codes, including to what extent
```

Comment:
The language here makes it unclear to me if this includes `kDif<ip>Ok`, `kDif<ip>Error`, and/or `kDif<ip>BadArg` or just the error codes belonging to `dif_<ip>_<operation>_result_t`.